src/org/zaproxy/zap/extension/invoke/InvokeAppWorker.java

```diff
@@ -101,7 +105,32 @@ protected Void doInBackground() throws Exception {
 						.replace("%port%", port)
 						.replace("%site%", site)
 						.replace("%cookie%", cookie)
-						.replace("%postdata%", postdata);
+						.replace("%postdata%", postdata)
+						.replace("%msgid%", String.valueOf(msgid));
+
+				// Replace header tags
+				Matcher headers = Pattern.compile("%header-([a-zA-z0-9_-]+)%").matcher(finalParameter);
+				while (headers.find()) {
+					String headerTag = headers.group(1);
+					String headerKey = headerTag;
+					String headerValue = "";
+
+					// If header tag is all lower case transform 
+					// user-agent into User-Agent as key for getting header value
+					if (headerKey.equals(headerKey.toLowerCase()))  {
```

Comment:
This is not necessary, `HttpHeader.getHeader(String)` already normalises the header name.